core/templates/dev/head/pages/topic_editor/questions/QuestionsTabDirective.js

```diff
@@ -118,6 +125,117 @@ oppia.directive('questionsTab', [
+                function(threadsResponse) {
+                  var suggestionThreads = (
+                    threadsResponse.data.suggestion_thread_dicts);
+                  var suggestions = suggestionsResponse.data.suggestions;
+                  if (suggestionThreads.length !== suggestions.length) {
+                    $log.error(
+                      'Number of suggestion threads doesn\'t match number of' +
+                       'suggestion objects');
+                  }
+                  for (var i = 0; i < suggestionThreads.length; i++) {
+                    for (var j = 0; j < suggestions.length; j++) {
+                      if (suggestionThreads[i].thread_id ===
+                          suggestions[j].suggestion_id) {
+                        var suggestionThread = (
+                          SuggestionThreadObjectFactory.createFromBackendDicts(
+                            suggestionThreads[i], suggestions[j]));
+                        $scope.questionSuggestionThreads.push(suggestionThread);
+                        break;
+                      }
+                    }
+                  }
+                });
+            });
+          };
+
+          $scope.setQuestionUnderReview = function(questionSuggestionThread) {
+            $scope.questionUnderReview = (
+              questionSuggestionThread.suggestion.question);
+            $scope.idOfSuggestionUnderReview = (
+              questionSuggestionThread.suggestion.suggestionId);
+          };
+
+          $scope.clearQuestionUnderReview = function() {
+            $scope.questionUnderReview = null;
+            $scope.suggestionReviewMessage = null;
+            $scope.idOfSuggestionUnderReview = null;
+          };
+
+          $scope.showSelectSkillModal = function() {
+            var allSkillSummaries = [];
+            allSkillSummaries = allSkillSummaries.concat(
+              $scope.topic.getUncategorizedSkillSummaries());
+            for (var i = 0; i < $scope.topic.getSubtopics().length; i++) {
+              var subtopic = $scope.topic.getSubtopics()[i];
+              allSkillSummaries = allSkillSummaries.concat(
+                subtopic.getSkillSummaries());
+            }
+            $uibModal.open({
+              templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                '/pages/topic_editor/questions/' +
+                'select_skill_modal_directive.html'),
+              backdrop: true,
+              controller: [
+                '$scope', '$uibModalInstance',
+                function($scope, $uibModalInstance) {
+                  $scope.selectedSkillId = null;
+                  $scope.skillSummaries = allSkillSummaries;
+
+                  $scope.selectSkill = function(skillId) {
+                    $scope.selectedSkillId = skillId;
+                  };
+
+                  $scope.done = function() {
+                    $uibModalInstance.close($scope.selectedSkillId);
```

Comment:
So what actually happens to the selected skill ID after the modal is closed? AFAICT it isn't being used anywhere...

Reply:
This was my bad. I fixed it now.